composables/transaction/transactionOffer.ts

```diff
@@ -17,35 +19,44 @@ export const OFFER_MINT_PRICE = 5e8
 
 export const BLOCKS_PER_DAY = 300 * 24 // 12sec /block --> 300blocks/hr
 
-async function execMakingOffer(item: ActionOffer, api, executeTransaction) {
+async function execMakingOffer(item: ActionOffer, api: ApiPromise, executeTransaction) {
   const { accountId } = useAuth()
   const nfts = Array.isArray(item.token) ? item.token : [item.token]
   const transactions = await Promise.all(
-    nfts.map(async ({ price, nftSn, collectionId, duration }) => {
-      const offerId = getOfferCollectionId(item.urlPrefix as Prefix)
-      const nextId = Number.parseInt(await generateId())
-      const create = api.tx.nfts.mint(
-        offerId,
-        nextId,
-        accountId.value,
-        {
-          mintPrice: String(OFFER_MINT_PRICE),
-        },
-      )
+    nfts.map(async ({ price, desiredItem, desiredCollectionId, duration, offeredItem: offeredSn }) => {
+      const offeredCollectionId = getOfferCollectionId(item.urlPrefix as Prefix)
+      let offeredItem = Number(offeredSn)
+
+      const transactions: SubmittableExtrinsic<'promise'>[] = []
+
+      if (!offeredItem) {
+        offeredItem = Number.parseInt(await generateId())
+        const create = api.tx.nfts.mint(
+          offeredCollectionId,
+          offeredItem,
+          accountId.value,
+          {
+            mintPrice: String(OFFER_MINT_PRICE),
+          },
+        )
+        transactions.push(create)
+      }
 
       const offer = api.tx.nfts.createSwap(
-        offerId,
-        nextId,
-        collectionId,
-        nftSn,
+        offeredCollectionId,
+        offeredItem,
+        desiredCollectionId,
+        desiredItem,
         {
           amount: Number(price) || 0,
           direction: 'Send',
         },
```

Comment:
Make this null instesd of object if price is not provided

Reply:
atm users can't crate an offer with 0 price

made these changes to make sure is not possible to call the transaction without a price  [c8157cf](https://github.com/kodadot/nft-gallery/pull/11204/commits/c8157cfd70be373e91ec3afcaf476509eb3c1c2a)

or you mean they should be able to , "free" offer ? 

<img src=https://github.com/user-attachments/assets/973da016-b515-4135-86ed-2392b5cb080e width=400 />

![CleanShot 2024-12-29 at 12 05 27@2x](https://github.com/user-attachments/assets/a6a19234-2c7b-4f3c-a155-6750fe15390f)